### PR TITLE
Feature/gh 8704 nextgen

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Loads of refactors and UX changes / optimisations.

### CODE REVIEW

File: feature-libs/my-account/organization/components/budget/budget.config.ts
Comment (Pucek9) [inline feature-libs/my-account/organization/components/budget/budget.config.ts]: Why did you decide to remove AdminGuard from all configs?

File: feature-libs/my-account/organization/assets/translations/en/user-group.i18n.ts
Comment (Pucek9) [inline feature-libs/my-account/organization/assets/translations/en/user-group.i18n.ts]: What will we do with those comments? Will these translations be useful in the future?

**Review by Pucek9 — CHANGES_REQUESTED** [review]
n/a

**Review by Pucek9 — CHANGES_REQUESTED** [review]
n/a

**Review by Pucek9 — CHANGES_REQUESTED** [review]
n/a

**Review by Pucek9 — CHANGES_REQUESTED** [review]
budgets, user groups and permissions list have link in first column and whole row has clickable cursor, for cost centers and users its missing

update: issue occurs in not specified situation, idk how to reproduce

**Review by Pucek9 — CHANGES_REQUESTED** [review]
n/a

File: feature-libs/my-account/organization/assets/translations/en/permission.i18n.ts
Comment (Pucek9) [inline feature-libs/my-account/organization/assets/translations/en/permission.i18n.ts:24]: [organization:permission.per.YEAR] is missing now, why did you remove that? We had also cases with empty timespan, it was simplified by using `undefined` key. I cant see alternative solution in `limit-cell.component.html`

File: feature-libs/my-account/organization/components/budget/budget.config.ts
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/components/budget/budget.config.ts]: I've overlooked that I'm afraid when I merged, thx! I'll put it back.

File: feature-libs/my-account/organization/assets/translations/en/user-group.i18n.ts
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/assets/translations/en/user-group.i18n.ts]: I've cleaned it up

File: feature-libs/my-account/organization/assets/translations/en/permission.i18n.ts
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/assets/translations/en/permission.i18n.ts:24]: I've added the the year back. 
I undefined issue is solved by only rendering the `periodRange` for `isTimeSpanThreshold` type.

Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/assets/translations/en/permission.i18n.ts:24]: Also fixed a bug to avoid overflow issue with the select.

**Comment by tobi-or-not-tobi** [comment]
> When I open `/cost-centers/${code}/budgets` there is header "MANAGE BUDGETS", and it should be "ASSIGNED BUDGETS", which is used vice versa in `cost-centers/${code}/budgets/assign`

The UX team want's to have "manage" rather than assign. It might change again going forward, and not sure if it will impact the actual route as well. Let's keep

**Review by Pucek9 — COMMENTED** [review]
n/a

**Comment by Pucek9** [comment]
units/${code}/users - missing roles headers in table

File: feature-libs/my-account/organization/assets/translations/en/cost-center.i18n.ts
Comment (Pucek9) [inline feature-libs/my-account/organization/assets/translations/en/cost-center.i18n.ts:49]:
```suggestion
  title: 'Assigned budgets',
```

File: feature-libs/my-account/organization/assets/translations/en/cost-center.i18n.ts
Comment (Pucek9) [inline feature-libs/my-account/organization/assets/translations/en/cost-center.i18n.ts:56]:
```suggestion
  title: 'Manage budgets',
```

File: feature-libs/my-account/organization/components/unit/units.config.ts
Comment (Pucek9) [inline feature-libs/my-account/organization/components/unit/units.config.ts:203]: units/${code}/users - missing roles headers in table
```suggestion
      options: {
        hideHeader: false,
```

File: feature-libs/my-account/organization/components/user/form/user-form.service.ts
Comment (Pucek9) [inline feature-libs/my-account/organization/components/user/form/user-form.service.ts:34]: user form -> "Add the user to approvers for the unit" checkbox should be visible only if approver role is checked. That mechanism seems as not working

File: feature-libs/my-account/organization/assets/translations/translations.ts
Comment (Pucek9) [inline feature-libs/my-account/organization/assets/translations/translations.ts:31]:
```suggestion
    'userGroupPermissions',
    'userGroupAssignedPermissions',
```

File: feature-libs/my-account/organization/assets/translations/en/cost-center.i18n.ts
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/assets/translations/en/cost-center.i18n.ts:49]: UX teams wants this to become "Manage". It might affect the route as well, but i'll hold with that bigger change for now, as these are smaller text changes.

File: projects/core/src/user/facade/user-address.service.ts
Comment (Pucek9) [inline projects/core/src/user/facade/user-address.service.ts]: unit tests needs to be updated after those changes

File: feature-libs/my-account/organization/components/unit/units.config.ts
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/components/unit/units.config.ts:203]: I'm holding off with changing this once more in this branch. We've received a new UX for the roles changes, which looks great, but I like to align with UX team once more.

File: feature-libs/my-account/organization/components/user/form/user-form.service.ts
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/components/user/form/user-form.service.ts:34]: Yes, we moved to a disabled state instead. The whole feature seems buggy, and we need to pick it up separately; doesn't seem to work on my company branch as well.

Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/components/user/form/user-form.service.ts:34]: I'm going to add a bit more content to #8418 to help the dev understanding the issue

File: projects/core/src/user/facade/user-address.service.ts
Comment (tobi-or-not-tobi) [reply projects/core/src/user/facade/user-address.service.ts]: Actually I'm going to revert this, as it's an existing core API.

**Review by Pucek9 — APPROVED** [review]

**Pull request merged**
